src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -77,6 +77,62 @@ internal static ParseResult ParseScript(string script, ILanguageSettings loc = n
             return new ParseResult(parsetree, errors, errors?.Any() ?? false, parser._comments, parser._before, parser._after);
         }
 
+        internal static ParseFormulasResult ParseFormulasScript(string script, ILanguageSettings loc = null, Flags flags = Flags.None)
+        {
+            Contracts.AssertValue(script);
+            Contracts.AssertValueOrNull(loc);
+
+            Dictionary<DName, ParseResult> namedFormulas = new Dictionary<DName, ParseResult>();
+            List<TexlError> errors = new List<TexlError>();
+
+            // Split formulas based on the semicolon
```

Comment:
I do not think that string.Split should appear anywhere in a parser. Was this approach recommended by somebody in particular?

A parser is a set of rules. Those rules can be re-used in other rules. For example you can have a rule expr() and a larger rule formula() and expr() is called inside formula()

Imagine the parser rule named powerfx(). You could re-use that rule in a larger rule, namedformulas() which calls powerfx() internally.

// Pseudocode
namedformula() {
   done = false
   while(!done)
   {
      identifier()
      equals()
      powerfx()
      if(!can_semicolon()) done = true
      else semicolon()
}